src/deion/worker/core/team/valueChange.js

```diff
@@ -54,22 +56,29 @@ const valueChange = async (
 		const fudgeFactor = (tid !== g.userTid ? 1.05 : 1) * difficultyFudgeFactor;
 
 		// Get roster and players to remove
-		const players = await idb.cache.players.indexGetAll("playersByTid", tid);
+		const players = await idb.cache.players.indexGetAll("playersByTid", tid); //get all the players on tid
 		for (const p of players) {
+			//for each player
+			updateValues(p);
 			if (!pidsRemove.includes(p.pid)) {
+				//player from tid not in pidsRemove put on roster
 				roster.push({
-					value: p.value,
+					value: strategy === "contending" ? p.valueNoPot : p.value,
```

Comment:
Seems like possibly too extreme a change, not sure. Contending teams certainly should value current performance more than rebuilding teams, but they shouldn't completely ignore potential.

Reply:
I found that adding the p.valueNoPot had the computer making trades like: 

team sends: average or good player + 1st or 2nd round pick
team get: good or great player

This is a realistic trade between a 'buyer' and 'seller' in the NBA. An example of this trade would be a fictional trade between the New Orleans Pelicans (NO) and the Miami Heat (MIA):

MIA gets: Jrue Holiday, E'Twaun Moore
MIA sends: Goran Dragic, James Johnson, and a bunch of future picks
(Moore and Johnson are there for salary matching)

MIA breakdown: we get our all-star to pair with Jimmy Butler (J. Holiday)
NO breakdown: gain value by clearing the books for the future, adding future assets (picks). [Both Dragic and Johnson are on 1 yr deals, Holiday is on a 3yr deal]

Switching between p.valueNoPot and p.value allows the computer to act like both MIA and NO. In the first scenario the computer is MIA and is looking to add a star right now. If I was the player or another computer controlled team, I could send a shitty player with a ton of potential to MIA and rob them an asset I want like Bam Adebayo. That's not a good deal for MIA. MIA is looking to swap a slight worse player + a future asset (contending teams don't care about the distant future) for a slightly better player. If the computer is now NO, we switch gears from win now to win later. NO has a couple of objectives -- build future assets, clear the books, and lose games. I can't have rebuilding teams actively search out bad players to lose games (Sam Hinkie won't be the CPU), but I can have them search for future assets and favorable short term contracts. P.value will lower dragic's value more than Holiday's, but valuing the contracts will balance the change in value (dragic - 1yr/$19M : holiday 3yr/$26M). The small difference in value will leave the 'selling' team wanting a little bit more. The buying team will happily send a future asset (a pick or two) for the player they think will make a difference right now. 

TL; DR: Using the ternary operator here creates a more realistic trade market between bad teams that I call 'sellers' and good teams that I call 'buyers'. It is hard to properly define a buyers motivations. A simplification of a buyer is ask the computer to strictly seek out talented players right now if contending/buying. Bad teams or sellers are looking to dump those assets for future assets. Using p.valueNoPot is strict, but in the team scope works in a friendly manner and allows for healthier trades.

Conclusion: 
I strongly believe that contending teams need  to use p.valueNoPot. I want my opinions to be heard, but you're the boss so if you want me to add a fractional amount of p.value to the statement or just remove p.valueNoPot, I will. I think the hardest part about accepting the p.valueNoPot is thinking about a player's value in the scope of the league vs. a player's value in the scope of a singular team. P.value is much more broad and more useful for league wide decisions, but the different variations of player value is more valuable in a team scope like valueChange.

![image](https://user-images.githubusercontent.com/54078512/72757978-49522c80-3b86-11ea-9614-d2083d28a6bc.png)

Reply:
This is the problem of not having tests, it's hard to debate. I'm just saying that, even for the greatest contender of all contenders, they will not consider a 20 year old and 35 year old player to have equal value just because their current abilities are equal. Maybe the formula you have now captures this well enough.